python/gui/auto_generated/qgsprovidersourcewidget.sip.in

```diff
@@ -38,6 +39,20 @@ Sets the source ``uri`` to show in the widget.
 Returns the source URI as currently defined by the widget.
 
 .. seealso:: :py:func:`setSourceUri`
+%End
+
+    void setMapCanvas( QgsMapCanvas *canvas );
```

Comment:
make virtual so that subclasses can implement special logic to set the canvas for their child widgets